components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -1699,6 +1699,10 @@ bool RewardsServiceImpl::ShouldShowOnboarding() const {
 void RewardsServiceImpl::EnableRewards() {
   StartProcess(base::BindOnce(
       &RewardsServiceImpl::OnStartProcessForEnableRewards, AsWeakPtr()));
+
+  for (auto& observer : observers_) {
+    observer.OnRewardsEnabled();
```

Comment:
This is not right, rewards is not enabled at this point because the process is async. To check for rewards/ads enabled you should use `prefs::kAutoContributeEnabled` and `ads::prefs::kEnabled` respectively with pref observers

Reply:
The code is a bit confusing and I'm not really sure what's going on here because there are different observers depending on how you enable rewards

Reply:
This is actually the point when user opted in the rewards. 
I think we can change OnRewardsEnabled to OnRewardsOptedIn to remove confusing points.